frontend/src/views/main/project/modals/CreateSliceModal.vue

```diff
@@ -0,0 +1,148 @@
+}>();
+
+const columnFilters = ref<Array<Partial<NoCodeFilter>>>([{}])
+
+const emit = defineEmits(['created']);
+
+const CLAUSE_BUILDERS: { [clause in NoCodeSlicingType]: (column: string, value: string) => string } = {
+    [NoCodeSlicingType.IS]: (column, value) => `giskard.slicing.slice.EqualTo(${column}, ${value})`,
+    [NoCodeSlicingType.IS_NOT]: (column, value) => `giskard.slicing.slice.NotEqualTo(${column}, ${value})`,
+    [NoCodeSlicingType.CONTAINS]: (column, value) => `giskard.slicing.slice.ContainsWord(${column}, ${value}, is_not=False)`,
+    [NoCodeSlicingType.DOES_NOT_CONTAINS]: (column, value) => `giskard.slicing.slice.ContainsWord(${column}, ${value}, is_not=True)`,
+    [NoCodeSlicingType.STARTS_WITH]: (column, value) => `giskard.slicing.slice.StartsWith(${column}, ${value})`,
+    [NoCodeSlicingType.ENDS_WITH]: (column, value) => `giskard.slicing.slice.EndsWith(${column}, ${value})`,
+    [NoCodeSlicingType.IS_EMPTY]: (column, _) => `giskard.slicing.slice.EqualTo(${column}, float('NaN'))`,
+    [NoCodeSlicingType.IS_NOT_EMPTY]: (column, _) => `giskard.slicing.slice.NotEqualTo(${column}, float('NaN'))`,
+}
+
+const CLAUSE_SYMBOLS: { [clause in NoCodeSlicingType]: string } = {
+    [NoCodeSlicingType.IS]: '==',
+    [NoCodeSlicingType.IS_NOT]: '!=',
+    [NoCodeSlicingType.CONTAINS]: 'contains',
+    [NoCodeSlicingType.DOES_NOT_CONTAINS]: 'does not contain',
+    [NoCodeSlicingType.STARTS_WITH]: 'startswith',
+    [NoCodeSlicingType.ENDS_WITH]: 'endswith',
+    [NoCodeSlicingType.IS_EMPTY]: 'is empty',
+    [NoCodeSlicingType.IS_NOT_EMPTY]: 'is not empty',
+}
+
+const NO_VALUES = [NoCodeSlicingType.IS_EMPTY, NoCodeSlicingType.IS_NOT_EMPTY]
+
+function escapePython(val?: string | number | null, column?: string): string {
+    if (column && val && typeof val === 'string') {
+        const type = props.dataset.columnDtypes[column];
+        if (type.startsWith('int')) {
+            return escapePython(parseInt(val, 10))
+        } else if (type.startsWith('float')) {
+            return escapePython(parseFloat(val))
+        }
+    }
+
+    if (typeof val === 'string') {
+        return `'${val.replace("'", "\\'")}'`
+    } else {
+        return val !== null && val !== undefined ? val.toString() : 'None'
+    }
+}
+
+function clauseCode(clause: NoCodeFilter) {
+    return CLAUSE_BUILDERS[clause.slicingType](escapePython(clause.column), escapePython(clause.value, clause.column))
+}
+
+function clausesCode() {
+    return `[${columnFilters.value.map((v) => clauseCode(v as NoCodeFilter)).join(', ')}]`
+}
+
+function clauseToString(clause: NoCodeFilter) {
+    return `${escapePython(clause.column)} ${CLAUSE_SYMBOLS[clause.slicingType]} ${NO_VALUES.includes(clause.slicingType) ? '' : escapePython(clause.value, clause.column)}`
+}
+
+async function save(close) {
+    const name = columnFilters.value.map((v) => clauseToString(v as NoCodeFilter)).join(' & ')
+
+    const dto: SlicingFunctionDTO = {
+        uuid: uuidv4(),
```

Comment:
along the line with the previous comment, UI should be sending what the user provides (`columnFilters` , `name`), the rest should be done on the backend